src/MLJ.jl

```diff
@@ -108,54 +95,39 @@ MMI.metadata_pkg.(
 
 MMI.metadata_model(
   EvoTreeRegressor,
-  input_scitype=Union{
-    MMI.Table(MMI.Continuous, MMI.Count, MMI.OrderedFactor),
-    AbstractMatrix{MMI.Continuous},
-  },
+  input_scitype=MMI.Table(MMI.Continuous, MMI.Count, MMI.OrderedFactor, MMI.Multiclass),
```

Comment:
Since you no longer support matrices I guess this is breaking, right?

And just to be sure, the treatment of `Multiclass` is properly invariant under relabelling? That is, the order of the classes makes no difference to outcomes? If not, then I suggest we rule out `MMI.Multiclass` and leave the responsibility of `Multiclass` -> `OrderedFactor` coercion to the user. 

I only ask, because a similar issue arose at DecisionTree.jl, where the ordering of categoricals matters,  but this was not flagged in docs.

Reply:
The adoption of the Tables's API in EvoTrees was notably motivated by the support for actual `Unordered` factor. These receive a particular treatment in the establishing split points, so yes, they are order invariant. The unordered factors were already supported under the Matrix since they can be simply converted to their numeric levels and treated like normal ordered numerics. 

Since I'd rather not making this a breaking change, I've added a conversion from Matrix input to tables to `MMI.fit`, and adjusted the input scitypes accordingly, along a test with a matrix input.